osf/models/notable_domain.py

```diff
@@ -30,8 +31,7 @@ def choices(cls):
     )
 
     def save(self, *args, **kwargs):
-        # Override this method to mark related content
-        # as spam or ham when reclassifying domain name
+        enqueue_task(reclassify_domain_references(self._id))
```

Comment:
check out this test fail:
```
tests/test_spam_mixin.py:244: in test_trusted_domain
    note=NotableDomain.Note.ASSUME_HAM_UNTIL_REPORTED,
/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/django/db/models/manager.py:85: in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/django/db/models/query.py:394: in create
    obj.save(force_insert=True, using=self.db)
osf/models/notable_domain.py:34: in save
    enqueue_task(reclassify_domain_references(self._id))
E   AttributeError: 'NotableDomain' object has no attribute '_id'
```
This PR should probably have it's own tests as well. Also you need the `reclassify_domain_references.s` with the `.s` to indicate it's the signature for the task.